### PR TITLE
Fixes Issue #200, Correcting case/default expression clause

### PR DESCRIPTION
**Pull request by anubmat**
Fixes issue #200 
The expression clause in switch case-clause and default clause included recursive references to expression scope.

### CODE REVIEW

**Comment by anubmat** [comment]
@zhengbli , @DanielRosenwasser , can you please review this PR?

File: TypeScript.YAML-tmLanguage [hunk 1]
```diff
@@ -37,15 +37,15 @@ repository:
...
 
   ternary-operator:
     begin: (\?)
     end: (:)
     patterns:
-    - include: '#ternary-expression-type' 
+    - include: '#expression-type' 
 
-  ternary-expression-type:
+  expression-type:
```
Comment (DanielRosenwasser) [inline TypeScript.YAML-tmLanguage]: II still don't understand why this is called `expression-type`. Can you give this a more accurate name?

File: TypeScript.YAML-tmLanguage [hunk 2]
```diff
@@ -466,12 +466,12 @@ repository:
 
   case-clause:
     name: case-clause.expr.ts
-    begin: '(?<!\.)\b(case|default)\b'
+    begin: '(?<!\.)\b(case|default(?=:))\b'
```
Comment (DanielRosenwasser) [inline TypeScript.YAML-tmLanguage]: What is the precedence here? Is this parsed as "`case` OR `default(?=:)`"? Do you only highlight `case` if it immediately followed by a colon? Can you test this out explicitly?

File: TypeScript.YAML-tmLanguage (hunk 1 above)
Comment (anubmat) [inline TypeScript.YAML-tmLanguage]: Can you suggest something. I do not like expression-type myself. Maybe calling this expression and the other expression  something like expression-group?

File: TypeScript.YAML-tmLanguage (hunk 2 above)
Comment (anubmat) [inline TypeScript.YAML-tmLanguage]: This means case or (default followed by a lookahead for colon). It does not catch "case:".

Comment (anubmat) [inline TypeScript.YAML-tmLanguage]: I have tested "export default" which was causing the problem. There are existing tests with switch statement using case and default - and those work as expected.

File: TypeScript.YAML-tmLanguage (hunk 1 above)
Comment (DanielRosenwasser) [inline TypeScript.YAML-tmLanguage]: This is only used in the true/false branch of a ternary expression, right? How about `ternary-expression-branch`?

Comment (anubmat) [inline TypeScript.YAML-tmLanguage]: Nopes, I just used it in case-clause because the expression in case clause had a recursive call for expression scope.

Comment (anubmat) [inline TypeScript.YAML-tmLanguage]: How about just expression-branch?

Comment (DanielRosenwasser) [inline TypeScript.YAML-tmLanguage]: The real problem is that switch statements are not expressions (nor are control statements). But I guess this is fine just to break the circularity.

**Comment by DanielRosenwasser** [comment]
:+1:

**Comment by anubmat** [comment]
Merging this in. Will discuss and fix expression scope-name in a later PR.

**Pull request merged**